main.go

```diff
@@ -127,15 +129,13 @@ func main() {
 		apiroots[i] = path.Join(root, appname, apiversion)
 	}
 
-	msgChannel := make(chan []byte, 100)
-	producer := &ProducerConfig{
-		Brokers: []string{kafkaBootstrap},
-		Topic:   metricsTopic,
-		Async:   true,
+	events := make(chan []byte, eventBuffer)
+	if kafkaBootstrap != "" {
```

Comment:
Do we know what happens to the events channel if this Producer function never gets started? Does the channel just fill up?

Reply:
yeah, I think we talked about this. It'll fill up until the app can't take anymore requests. Nothing reads from it without the producer. This shouldn't effect anything live, but it does impact local testing if you don't have a producer running. You'd have to set the buffer high enough that you wont' hit the ceiling. I'm open to ideas here though

Reply:
Ok. I was able to verify this will block HTTP requests if the channel is full or unbuffered. That's not a good condition to release into the wild. If the broker dies, or the topic stops accepting new messages, the writer goroutine will just keep stuffing the current message back onto the channel, eventually filling it up. Once it's full, inbound HTTP requests will still accept, but they will never respond. Clients *should* eventually time out, but that will create undue lag in a client checking for payload updates.

Reply:
I resolved this by storing a `*chan []byte` in the `Server` instead. Then I switched to checking if that field is `nil` during `handleEvent`. If it is not `nil`, send the data to the buffer.